libyul/optimiser/StackToMemoryMover.cpp

```diff
@@ -58,28 +58,33 @@ void StackToMemoryMover::run(
 	uint64_t _numRequiredSlots,
 	Block& _block
 )
-{
-	VariableMemoryOffsetTracker memoryOffsetTracker(_reservedMemory, _memorySlots, _numRequiredSlots);
-	StackToMemoryMover stackToMemoryMover(_context, memoryOffsetTracker);
-	stackToMemoryMover(_block);
-}
-
-StackToMemoryMover::StackToMemoryMover(
-	OptimiserStepContext& _context,
-	VariableMemoryOffsetTracker const& _memoryOffsetTracker
-):
-m_context(_context),
-m_memoryOffsetTracker(_memoryOffsetTracker),
-m_nameDispenser(_context.dispenser)
 {
 	auto const* evmDialect = dynamic_cast<EVMDialect const*>(&_context.dialect);
 	yulAssert(
 		evmDialect && evmDialect->providesObjectAccess(),
 		"StackToMemoryMover can only be run on objects using the EVMDialect with object access."
 	);
+	VariableMemoryOffsetTracker memoryOffsetTracker(_reservedMemory, _memorySlots, _numRequiredSlots);
+
+	VariableDeclarationAndAssignmentMover declarationAndAssignmentMover(_context, memoryOffsetTracker);
+	declarationAndAssignmentMover(_block);
+
+	IdentifierMover identifierMover(_context, memoryOffsetTracker);
+	identifierMover(_block);
+}
+
+void StackToMemoryMover::VariableDeclarationAndAssignmentMover::operator()(FunctionDefinition& _functionDefinition)
```

Comment:
This is a duplicate of ``StackToMemoryMover::IdentifierMover::operator()`` - but both will disappear in the next step(s).